asynciterator.js

```diff
@@ -209,24 +220,63 @@ AsyncIterator.prototype.close = function () {
     endAsync(this);
 };
 
+/**
+ Destroy the iterator and stop it from generating new items.
+
+ This will not do anything if the iterator was already ended or destroyed.
+
+ All internal resources will be released an no new items will be emitted,
+ even not already generated items.
+
+ Implementors should not override this method,
+ but instead implement {@link AsyncIterator#_destroy}.
+
+ @param {Error} [error] An optional error to emit.
+ @emits AsyncIterator.end
+ @emits AsyncIterator.error Only emitted if an error is passed.
+ **/
+AsyncIterator.prototype.destroy = function (error) {
+  if (!this.done) {
+    var self = this;
+    this._destroy(error, function (errorInner) {
+      error = error || errorInner;
+      if (error)
+        self.emit('error', error);
+      end(self, true);
+    });
+  }
+};
+
+/**
+ Called by {@link AsyncIterator#destroy}.
+ Implementors can override this, but this should not be called directly.
+
+ @param {Error} error A possible error.
+ @param {Function} callback A callback function with an optional error argument.
+ */
+AsyncIterator.prototype._destroy = function (error, callback) {
```

Comment:
Why would this one take an error as argument?

Reply:
This is the error that was passed to `destroy`. This may be useful for implementors of `_destroy` to know if the stream was made to error or not.

This is how it's done in Node's `ReadableStream` was well.